### ISSUE TITLE
Build failed for python 3.6

### ISSUE THREAD

**Issue by phagunbaya** [Shippable/support#4623]
Moved from BYON to on-demand node recently. After that python project build failed.
https://app.shippable.com/github/Falkonry/kestrel/runs/656/1/console

```
virtualenv -p /usr/bin/python3.6 /root/venv/3.6
The path /usr/bin/python3.6 (from --python=/usr/bin/python3.6) does not exist
```

**Comment by rageshkrishna** [comment]
The node pool that was created for you automatically uses the latest available runtime version (v6.9.4), which does not have Python 3.6 available. It only has 3.7.

You can use runtime version v6.6.4 instead, which has Python 3.6.5:
http://docs.shippable.com/platform/runtime/machine-image/ami-v664/#python

You can pick the runtime version from the "Edit Node Pool" page: http://docs.shippable.com/platform/tutorial/runtime/manage-node-pools/#updating-node-pools

**Comment by rageshkrishna** [comment]
I see you got a green build on this. I'm closing the issue as it appears to be resolved. Feel free to reopen if you still have trouble.

**Comment by phagunbaya** [comment]
@rageshkrishna I want to downgrade the agent version on custom nodes (BYON). How to do that ?

**Comment by trriplejay** [comment]
@phagunbaya The agent can't be downgraded, but you can always modify the selected runtime version of your node pool.  For custom nodes, this mainly impacts the default build images used. So you'd probably want your BYON pool to be set to v6.6.4 as Ragesh suggested above.

**Comment by phagunbaya** [comment]
I want to go back to version 6.1.4. Anything above this version is breaking our builds.

**Comment by phagunbaya** [comment]
@rageshkrishna could you please help me with this. We cannot upgrade node version at this point in time. We are blocked on this.

**Comment by rageshkrishna** [comment]
@phagunbaya I don't see any BYON pools in your subscription. Can you please link me to the node pool that you are having trouble with?

**Comment by phagunbaya** [comment]
@rageshkrishna This is regarding the on-demand pool. I deleted the BYON. I want to downgrade the on-demand pool to version 6.1.4.

**Comment by rageshkrishna** [comment]
Thanks for that, I understand the problem now. v6.1.4 is not available with Ubuntu 16.04. 

To use v6.1.4, you will need to change your SKUs on the billing page to switch to Ubuntu 14.04. Just remove the existing SKU, add and configure the Ubuntu 14.04 SKU and save changes.

Once you do this, you will have a new default node pool set up for Ubuntu 14.04 and you will see that v6.1.4 is available for you to switch to.

Let me know if you have any trouble setting this up.

**Comment by rageshkrishna** [comment]
@phagunbaya It looks like you were able to set this up successfully. Please open a new issue if you need help with anything.

**Comment by phagunbaya** [comment]
Seems to be resolved. Thanks. Now my worry is why shippable is releasing incompatible versions.

**Comment by rageshkrishna** [comment]
The reason why v6.1.4 is not available with Ubuntu 16.04 is because our support for Ubuntu 16 was released only in v6.3.4. 

Besides that, we have documented what is available in each runtime version at http://docs.shippable.com/platform/runtime/machine-image/ami-overview/. Typically, newer versions of the runtime will have newer versions of JDKs, python versions, etc., so it's up to you to decide which versions work for your projects.

Does that clarify the situation, or did you mean something else by "incompatible versions"?
